### PR TITLE
Reorganise reader activation wizard with ActionCard/Toggle

### PR DESCRIPTION
**Pull request by thomasguillot**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Instead of using a section header then a toggle, I combined the two in an ActionCard/Toggle card.

Screenshot 2022-07-29 at 16 43 07

### How to test the changes in this Pull Request:

1. Check the reader activation wizard
2. Switch to this branch
3. Refresh wizard

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by dkoo — APPROVED** [review]
This works fine, so approving. I have two non-blocking suggestions, in order of ascending difficulty:

1. If Reader Activation is disabled in the action card, should we hide the rest of the settings?
2. Why not refactor to use the `PluginSettings` component we're using across other wizard pages, such as in the Advertising > Settings and Campaigns > Settings pages? e.g.

Screen Shot 2022-08-02 at 11 51 05 AM

IMO this component is well-suited for presenting the options here, and using it would maintain consistency across wizards.

**Comment by thomasguillot** [comment]
> Why not refactor to use the `PluginSettings` component

@dkoo It didn't even cross my mind 🤦‍♂️
Since it's not really a plugin, I couldn't use `PluginSettings` but I sort of replicated the style in https://github.com/Automattic/newspack-plugin/pull/1817/commits/b621bb9aa789c705b3865635effa667678b89626

Is this any better?

File: composer.json [hunk 1]
```diff
@@ -36,5 +36,11 @@
 				"cat $1 | ./node_modules/.bin/newspack-scripts commitlint"
 			]
 		}
+	},
+	"config": {
```
Comment (dkoo) [inline composer.json:40]: Is this change required for this PR? If not, we should probably handle this separately.

File: assets/wizards/engagement/views/reader-activation/index.js [hunk 2]
```diff
@@ -74,56 +73,51 @@ export default withWizardScreen( () => {
...
-					/>
-					<TextControl
-						label={ __( 'Terms & Conditions URL', 'newspack' ) }
-						help={ __( 'URL to the page containing the terms and conditions.', 'newspack' ) }
-						value={ config.terms_url }
-						onChange={ value => updateConfig( 'terms_url', value ) }
-					/>
-				</Grid>
-			</Card>
-			{ isActiveCampaign && (
```
Comment (dkoo) [inline assets/wizards/engagement/views/reader-activation/index.js:116]: Did you mean to remove the ActiveCampaign master list settings?

**Review by dkoo — CHANGES_REQUESTED** [review]
This looks great! You can actually use the `PluginSettings` component as any regular component, it just requires a very specific data shape and API endpoints, so your solution to use other components to match the design works for me.

Some lingering questions below.

File: assets/wizards/engagement/views/reader-activation/index.js (hunk 2 above)
Comment (thomasguillot) [reply assets/wizards/engagement/views/reader-activation/index.js:116]: The rebase clearly didn't go well 😅
I'll open a new PR, it'll be easier

Comment (thomasguillot) [reply assets/wizards/engagement/views/reader-activation/index.js:116]: https://github.com/Automattic/newspack-plugin/pull/1852

**Comment by thomasguillot** [comment]
Closing in favour of https://github.com/Automattic/newspack-plugin/pull/1852

File: composer.json (hunk 1 above)
Comment (thomasguillot) [reply composer.json:40]: Yep, mistake on my end.
